### PR TITLE
Update_week_one.md

### PR DESCRIPTION
**Pull request by dijital2016**

### CODE REVIEW

File: module-2/backend/week_one.md [hunk 1]
```diff
@@ -5,28 +5,76 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
   
   ```ruby
   get '/horses' do
+  @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
+name = "Mr. Ed"
```
Comment (case-eee) [inline module-2/backend/week_one.md]: the local variable `name` would not be available within the view template if you don't explicitly follow a specific syntax like so:

```ruby
get '/horses' do
  erb :index, locals: {name: "Mr. Ed"}
end
```

File: module-2/backend/week_one.md [hunk 2]
```diff
@@ -5,28 +5,76 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+Hyper-text Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+We can use regular interpolation or compact interpolation which is not often used and somewhat controversial.
+
 15. What's an ORM?
+Object Relational mapping, such as ActiveRecord which wraps SQL commands in code.
+
 16. What's the most commonly used ORM?
+Had to look this one up.  Evidently it's Hibernate which uses Java.
```
Comment (case-eee) [inline module-2/backend/week_one.md]: Bad question - I should have specified the most commonly used ORM in Ruby - which is ActiveRecord 👍

File: module-2/backend/week_one.md [hunk 3]
```diff
@@ -5,28 +5,76 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+GET '/index' --See all stored items. (Read)
```
Comment (case-eee) [inline module-2/backend/week_one.md]: your http verbs and the purpose of each route is correct, the convention for the pathname is not. if we have a `restaurants` table, we would want the following:

```
get '/restaurants'
get '/restaurants/:id'
get '/restaurants/new'
post '/restaurants'
get '/restaurants/:id/edit'
put '/restaurants/:id'
delete '/restaurants/:id'
```

File: module-2/backend/week_one.md [hunk 4]
```diff
@@ -5,28 +5,76 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+DELETE '/item' Delete a specific item from the database (Delete)
+
 18. What's a migration? 
+A part of the process of transferring data into a database.
+
 19. When you create a migration, does it automatically modify your database?
+No.  First you create the migration.  Then you execute the migration in a separate step.
+
 20. How does a model relate to a database?
+It keeps all the code for a specific function.
```
Comment (case-eee) [inline module-2/backend/week_one.md]: not quite sure what you mean by this. could you be more specific?

**Comment by case-eee** [comment]
hey @dijital2016 i left ya some comments. please let me know if you have any questions 👍 keep up the hard work!

File: module-2/backend/week_two.md [hunk 5]
```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+---It's a wrapper for SQL that allows you to easily access a database within an application with easy commands.
+
 2. What is a migration?
+--It's a set of instructions for creating a database.
+
 3. How does a table relate to a model?
+--We have a model for each table.  A model is a class that does calculations and operations on the data.
+
 4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+--They are class methods, which is why they're in the models which are classes.  In Blogger, we had an Article class and an Attachment class.  Each attachment "belonged_to" an article and an article could "have_many" attachments.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: Tricky question: these are actually instance methods. The important thing to remember is that you need a specific instance before you can use these. For example:

```ruby
kitten_in_tree = Article.find(1) # assigns the Article with an `id` of 1 to the variable `kitten_in_tree`.
attached_image = kitten_in_tree.attachment # uses the `attachment` method that `has_many :attachments` gives us in our Article class.
```

File: module-2/backend/week_two.md [hunk 6]
```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
 5. What do they allow you to do?
+--They establish relationships between tables and classes.
+
 6. What's the difference between agile workflow and waterfall method?
+--Agile workflow starts with small chunks of functionality and tests and reevaluates each step along the way.  Waterfall does all the planning at first with end goals in mind, then builds it all out, and then tests it all at the end.
+
 7. What is the difference between `#new` and `#create`?
+--Both create a new object but only create saves it in the database.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 👍

File: module-2/backend/week_two.md [hunk 7]
```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+--The primary key is the main identifier in a table.  A foreign key is usually an id from a different table which relates the two.  A schema is basically a map of a database and the relationships between its tables.
+
 11. Describe the relationship between a foreign key on one table and a primary key on another table.
+--The foreign key in one table refers to the primary key of another table and establishes a relationship between them.
+
 12. What are the parts of an HTTP response?
+--There is a code (3 digits), head, and body.
+
 13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
+--Assert and assert_equal
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: Think we dropped this, but then didn't remind you to pull the latest version. We didn't teach about controller tests this time around.

File: module-2/backend/week_two.md [hunk 8]
```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
 14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+--If you find yourself repeating the same code over and over, such as creating Payloads in RushHour, either use something like FactoryGirl or simply create your own method to create payloads and simply call the method each test.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: Good ideas for drying up code in general, but in terms of **views** specifically, I would take a look at partials. We'll get into this more tomorrow.

File: module-2/backend/week_two.md [hunk 9]
```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+---It's a wrapper for SQL that allows you to easily access a database within an application with easy commands.
+
 2. What is a migration?
+--It's a set of instructions for creating a database.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 👍

File: module-2/backend/week_two.md [hunk 10]
```diff
@@ -5,19 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+--Both create a new object but only create saves it in the database.
+
 8. At a basic level, what does cURL allow you to do?
+--It enables you to communicate with servers over the internet from the command line or from within a program.
+
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+--Many-to-Many.  There will be a students table and a teachers table, each of which are connected through a joined students_teachers table.
+
 10. Define foreign key, primary key, and schema.
+--The primary key is the main identifier in a table.  A foreign key is usually an id from a different table which relates the two.  A schema is basically a map of a database and the relationships between its tables.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 🎉

**Pull request closed without merging**